test/jasmine/tests/image_test.js

```diff
@@ -57,6 +60,37 @@ describe('image supplyDefaults', function() {
         expect(traceOut.visible).toBe(true);
     });
 
+    it('should set visible to false when source is empty', function() {
+        traceIn = {
+            source: null
+        };
+        supplyDefaults(traceIn, traceOut);
+        expect(traceOut.visible).toBe(false);
+
+        traceIn = {
+            type: 'image',
+            source: 'data:image/png;base64,somedata'
+        };
+        traceOut = Plots.supplyTraceDefaults(traceIn, {type: 'image'}, 0, layout);
+        expect(traceOut.visible).toBe(true);
+    });
+
+    it('should set visible to false when source is a URL', function() {
+        traceIn = {
+            source: 'https://antrg.com/assets/img/portrait_2013_circle_200px.png'
+        };
+        supplyDefaults(traceIn, traceOut);
+        expect(traceOut.visible).toBe(false);
+        expect(traceOut.source).toBe(undefined);
+
+        traceIn = {
+            type: 'image',
+            source: 'data:image/png;base64,somedata'
+        };
+        traceOut = Plots.supplyTraceDefaults(traceIn, {type: 'image'}, 0, layout);
+        expect(traceOut.visible).toBe(true);
+    });
+
```

Comment:
Could you please add tests to reject cases similar to XXS attacks similar to
https://github.com/plotly/plotly.js/blob/3781f482455e7623d1a0368b6e74427df549550b/test/jasmine/tests/svg_text_utils_test.js#L516

Reply:
> Apparently IE11 has issues with `onload` handler for images and that's a showstopper. Investigating a workaround.
> 
> cc @nicolaskruchten @archmoj @alexcjohnson

Safari has a similar issue to IE with `onload` but fortunately it seems restricted to `<image>` (an SVG element) and not `<img>` (an HTML element). I got this information from comments in this [Stackoverflow thread](https://stackoverflow.com/questions/35036672/trap-the-image-loading-event-on-an-svg-element-in-ie11).

Reply:
Done in fe1d2083c4ae61d7c9ee85561fb4654f4cde4643